example/test/avoid_non_null_assertion_test.dart

```diff
@@ -0,0 +1,22 @@
+/// Check "bang" operator fail
+///
+/// `avoid_non_null_assertion`
+class AvoidNonNullAssertion {
+  AvoidNonNullAssertion? object;
+  int? number;
+
+  void test() {
+    // expect_lint: avoid_non_null_assertion
+    number!;
+
+    // expect_lint: avoid_non_null_assertion
+    object!.number!;
+
+    // expect_lint: avoid_non_null_assertion
+    object!.test();
+
+    // No lint on maps
+    final map = {'key': 'value'};
+    map['key']!;
```

Comment:
This might be debatable. It was implemented this way in DCM: https://github.com/dart-code-checker/dart-code-metrics/issues/1015.

However, we might want to develop a fix. I'll open an issue for it.